addons/mail/static/src/models/message_action_list.js

```diff
@@ -1,187 +1,162 @@
-         * @returns {boolean}
+         * @returns {FieldCommand}
          */
-        _computeHasCompactIcon() {
-            const COMPACT_THRESHOLD = 2;
-            const viewer = this.messageView && this.messageView.messageListViewMessageViewItemOwner && this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.threadViewer;
-            if (viewer && viewer.chatWindow) {
-                return this.actionsCount > COMPACT_THRESHOLD;
+        _computeReplyToActionView() {
+            if (this.hasReplyToAction) {
+                return insertAndReplace();
             }
+            return clear();
         },
         /**
          * @private
-         * @returns {boolean}
+         * @returns {FieldCommand}
          */
-        _computeHasMarkAsReadIcon() {
-            return Boolean(
-                this.messaging && this.messaging.inbox &&
-                this.messageView && this.messageView.messageListViewMessageViewItemOwner && this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread &&
-                this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread === this.messaging.inbox
-            );
+        _computeToggleCompactActionView() {
+            if (this.messageView.isInChatWindow && (this.actionsWithoutCompactCount > this.compactThreshold)) {
+                return insertAndReplace();
+            }
+            return clear();
         },
         /**
          * @private
-         * @returns {boolean}
+         * @returns {FieldCommand}
          */
-        _computeHasReplyIcon() {
-            return Boolean(
-                this.messaging && this.messaging.inbox &&
-                this.message && !this.message.isTemporary && !this.message.isTransient &&
-                this.messageView && this.messageView.messageListViewMessageViewItemOwner && this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread && (
-                    this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread === this.messaging.inbox ||
-                    this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread.model === 'mail.channel'
-                )
-            );
+        _computeToggleStarActionView() {
+            if (this.message && this.message.canStarBeToggled) {
+                return insertAndReplace();
+            }
+            return clear();
         },
+        _sortMessageActionViews() {
+            return [
+                ['smaller-first', 'sequence'],
+            ];
+        }
     },
     fields: {
-        /**
-         * States the reference to the reaction action in the component.
-         */
-        actionReactionRef: attr(),
         actionsCount: attr({
-            compute: '_computeActionsCount',
             readonly: true,
+            sum: 'messageActionViews',
```

Comment:
Is that actually working? I don't think we implemented "length" into that feature, isn't it supposed to target a field on the related model that provides the value (in this case default: 1)

Reply:
nice to know, now it works...